compiler/optimizations/gpuTransforms.cpp

```diff
@@ -316,14 +316,46 @@ static  CallExpr* generateGPUCall(OutlineInfo& info, VarSymbol* numThreads) {
   return call;
 }
 
+static void errorForOuterVarAccesses(FnSymbol* fn) {
+  std::vector<SymExpr*> ses;
+  collectSymExprs(fn, ses);
+  for_vector(SymExpr, se, ses) {
+    if (VarSymbol* var = toVarSymbol(se->symbol())) {
+      if (var->defPoint->parentSymbol != fn) {
+        if (!var->isParameter() && var != gVoid) {
+          USR_FATAL(se, "variable '%s' must be defined in the function it"
+                    " is used in for GPU usage", var->name);
```

Comment:
Why do we error here? Can't we just silently ignore the loop where a call has outer variable access?